web-common/src/features/entity-management/file-selectors.ts

```diff
@@ -153,25 +160,52 @@ export function useFileNamesInDirectory(
   instanceId: string,
   directoryPath: string,
 ) {
-  directoryPath += "/";
+  // Ensure the directory path starts with a slash
+  if (!directoryPath.startsWith("/")) {
+    directoryPath = `/${directoryPath}`;
+  }
+
   return createRuntimeServiceListFiles(instanceId, undefined, {
     query: {
       select: (data) => {
-        const files = data.files?.filter(
-          (file) => !file.isDir && file.path?.startsWith(directoryPath),
-        );
-        const fileNames = files
-          ?.map((file) => {
-            return file.path?.replace(directoryPath, "") ?? "";
+        if (!data.files) {
```

Comment:
Would be good to add tests to these. Perhaps just this selector function? Same with the selector function in `useDirectoryNamesInDirectory`